flake.nix

```diff
@@ -0,0 +1,82 @@
+{
+    description = "Snowbridge flake";
+
+    inputs = {
+        nixpkgs.url = "nixpkgs/nixos-22.11";
+        rust-overlay.url = "github:oxalica/rust-overlay";
+        flake-utils.url  = "github:numtide/flake-utils";
+    };
+
+    outputs = { self, nixpkgs, rust-overlay, flake-utils }:
+    let
+        supportedSystems = [ "aarch64-darwin" "x86_64-darwin" "x86_64-linux" ];
+        overlays = [ (import rust-overlay) ];
+    in
+    flake-utils.lib.eachSystem supportedSystems (system:
+        let
+            pkgs = import nixpkgs { inherit system overlays; };
+        in
+        with pkgs;
+        {
+            devShells.default = pkgs.mkShell {
+                buildInputs = [
+                    git
+                    curl
+                    jq
+                    moreutils
+                    direnv
+                    typos
+                    go-ethereum
+                    cacert
+
+                    # typescript packages
+                    nodejs-18_x
+                    nodePackages.pnpm
+
+                    # relayer
+                    go
+                    mage
+                    revive
+
+                    # parachain
+                    rustup
+
+                    gcc
+                    libiconv
+                    protobuf
+
+                    clang
+                    cmake
+                    openssl
+
+                    cowsay
+                ];
+
+                shellHook = ''
+                    # set HOME for direnv and go
+                    #
+                    # direnv needs config, cache & data dirs (DIRENV_CONFIG, XDG_CACHE_HOME & XDG_DATA_HOME
+                    # respectively) that can be automatically set when HOME is available
+                    #
+                    # relayer builds fail without GOPATH & GOCACHE set
+                    # explicitly setting HOME allows go to infer these vars
+                    #
+                    export HOME=~
+
+                    eval "$(direnv hook bash)"
+
+                    # rocksdb requires a clang.so
+                    export LIBCLANG_PATH="$(readlink -f ${pkgs.clang}/resource-root/include | xargs dirname | xargs dirname | xargs dirname)"
+
+                    pushd parachain
+                    cargo install --git https://github.com/paritytech/polkadot --tag v0.9.30 polkadot --locked --root .cargo
```

Comment:
Won't this time-consuming process cause the shell to hang every time its activated?

Reply:
Also its fine for the installation of polkadot to remain the responsibility of start-services.

Nice trick with the `cargo install` approach though. Much simpler than cloning the polkadot repo, so am keen to retain that part.

Reply:
If the binary exists, `cargo install` skips it. This only happens on the first `nix develop` invocation, and whenever the binary is removed, eg. accidentally running `git clean` without filtering it out. You can also copy another version of a binary into `parachain/.cargo/bin/polkadot` if you need to test other versions.

Happy to move this into `start-services`, it is a bit annoying when the binary doesn't exist.

Reply:
Ah ok, cool. Well if cargo install skips it then it's fine to put the install in the shellhook